components/sup/src/manager/service/config.rs

```diff
@@ -227,6 +234,18 @@ impl ServiceConfig {
         self.needs_write = false;
         Ok(should_restart)
     }
+
+    pub fn reload_package(&mut self, package: &PackageInstall) -> Result<()> {
+        self.config_root = match self.config_root_from_config {
+            Some(ref root) => root.clone(),
+            None => package.installed_path.clone(),
+        };
+        self.pkg = Pkg::new(package,
+                            self.pkg.svc_user.clone(),
+                            self.pkg.svc_group.clone())?;
+        self.cfg = Cfg::new(package, &self.config_root)?;
+        Ok(())
```

Comment:
Oops! I meant to leave this comment on `update_package` on the `Service` struct and not this function on `ServiceConfig`. I see that you've defined some new functions for recompiling the templates but what I am suggesting is to completely throw away the old `HooksTable` and reload it